apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -112,40 +137,62 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
             offset: state.offset,
             pageSize: state.pageSize,
         })
-        const meterReadings = await readMeterReadings({ context, meters })
         const organizations = await readOrganizations({ context, meters })
+        const reportingPeriods = await readMeterReportingPeriods({ context, organizations })
+        const meterReadings = await readMeterReadings({ context, meters, reportingPeriods })
+
+        const metersWithoutReadings = []
+        const periodsByProperty = []
+        const periodsByOrganization = []
+        let defaultPeriod = null
+
+        for (let period of reportingPeriods) {
+            if (!period.organization) defaultPeriod = period
+            else if (!period.property) periodsByOrganization.push(period)
+            else periodsByProperty.push(period)
+        }
 
-        // right join data to metersPage
-        const metersWithoutReadings = rightJoin(
-            meters,
-            meterReadings,
-            (meter, item) => meter.id === item.meter.id,
-            (meter, readings) => ({ meter, readings })
-        )
-            .filter(({ readings }) => readings.length === 0)
-            .map(({ meter }) => meter)
+        for (let meter of meters) {
```

Comment:
Why not const? I can't see any mutation of array elements inside of the loop
```suggestion
        for (const meter of meters) {
```